evap/staff/tools.py

```diff
@@ -65,6 +66,15 @@ def create_user_list_html_string_for_message(users):
     return format_html_join("", "<br />{} {} ({})", ((user.first_name, user.last_name, user.email) for user in users))
 
 
+def append_user_list_if_not_empty(message: str, user_profiles: Collection) -> str:
+    if not user_profiles:
+        return _("{message}.").format(message=message)
+    return _("{message}: {list}").format(
+        message=message,
+        list=create_user_list_html_string_for_message(user_profiles),
+    )
```

Comment:
Richard and I talked about this some more and we came to the conclusion that we generally don't actually always want `format_html` if there is nothing in the format string that has to explicitly not-escaped. So instead, a concatenation of `SafeString` is best. Further, we can ensure that we are not loosing the escaping on the way by using type annotations:
```python
from django.utils.safestring import SafeString
from django.utils.html import conditional_escape, escape


def append_user_list_if_not_empty(message: str, user_profiles: Collection) -> SafeString:
    message = conditional_escape(message)
    assert isinstance(message, SafeString)
    if not user_profiles:
        return message + escape(".")
    return message + escape(": ") + create_user_list_html_string_for_message(user_profiles)
```
Please make sure to also annotate `create_user_list_html_string_for_message` so that it returns `SafeString` (and takes `Iterable[UserProfile]`)

Reply:
And actually, we came up with another thing: Django's `conditional_escape` is annotated to only return `str`, not `SafeString`. This stems from the fact that Django tries to be compatible with other escaping libraries. However, we only care about `str` and `SafeString`. Let's write our own `conditional_escape` in `evaluation/tools.py` like so:
```python
def conditional_escape(s: str) -> SafeString:
    """
    Like Django's `conditional_escape`, but only handles `str` and `SafeString`.
    """
    if isinstance(s, SafeString):
        return s
    return escape(s)
```
Using this, we can remove the `assert` from the code in my last comment